algorithms/linfa-clustering/src/optics/hyperparameters.rs

```diff
@@ -0,0 +1,69 @@
+use crate::optics::errors::{OpticsError, Result};
+#[cfg(feature = "serde")]
+use serde_crate::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(crate = "serde_crate")
+)]
+/// The set of hyperparameters that can be specified for the execution of
+/// the [OPTICS algorithm](struct.Optics.html).
+pub struct OpticsHyperParams {
+    /// Distance between points for them to be considered neighbours.
+    tolerance: f64,
+    /// Minimum number of neighboring points a point needs to have to be a core
+    /// point and not a noise point.
+    min_points: usize,
+}
+
+impl OpticsHyperParams {
+    /// Minimum number of neighboring points a point needs to have to be a core
+    /// point and not a noise point.
+    ///
+    /// Defaults are provided if the optional parameters are not specified:
+    /// * `tolerance = f64::MAX`
+    pub fn new(min_points: usize) -> Self {
+        OpticsHyperParams {
+            min_points,
+            tolerance: std::f64::MAX,
+        }
+    }
+
+    /// Distance between points for them to be considered neighbors. Compared to DBSCAN this
+    /// parameter isn't strictly necessary but improves execution time by not considering every
+    /// point. If the tolerance is too low the distances calculated are undefined and no clusters
+    /// will be returned.
+    pub fn with_tolerance(mut self, tolerance: f64) -> Self {
+        self.tolerance = tolerance;
+        self
+    }
+
+    /// Two points are considered neighbors if the euclidean distance between
+    /// them is below the tolerance
+    pub fn tolerance(&self) -> f64 {
+        self.tolerance
+    }
```

Comment:
Can you rename these methods to `tolerance` and `get_tolerance` respectively? `with_tolerance` seems like a weird name for a builder method.

Reply:
Sure, I just did it to match the new lazy builder pattern using the `gaussian_mixture` module as reference, so someone might want to do a PR to change all the `with_*` methods there.